entropy/coin_flip/app/src/flip_coin.ts

```diff
@@ -29,75 +45,80 @@ const parser = yargs(hideBin(process.argv))
 
 async function main() {
   const argv = await parser.argv;
+  const chainName = argv.chainName;
+  if (chainName !== "optimism-sepolia" && chainName !== "arbitrum-sepolia") {
+    throw new Error("Invalid chain name");
+  }
+
+  let chain: Chain;
+  if (chainName === "optimism-sepolia") {
+    chain = optimismSepolia;
+  } else {
+    chain = arbitrumSepolia;
+  }
+
+  const client = createWalletClient({
+    chain: chain,
+    account: privateKeyToAccount(argv.privateKey as Hex),
+    transport: http(argv.rpcUrl),
+  }).extend(publicActions);
+
+  const coinFlipContract = getContract({
+    address: argv.address as Hex,
+    abi: ICoinFlipAbi,
+    client: client,
+  });
+
+  console.log("1. Generating user's random number...");
 
-  const coinFlipContractAddress = argv.address;
-  const rpc = argv.rpcUrl;
-  const privateKey = argv.privateKey;
+  const randomNumber = `0x${crypto.randomBytes(32).toString("hex")}`;
+  console.log(`User Generated Random number: ${randomNumber}`);
 
-  const provider = new HDWalletProvider({
-    privateKeys: [privateKey],
-    providerOrUrl: rpc,
+  console.log("\n2. Requesting coin flip...");
+
+  const flipFee = await coinFlipContract.read.getFlipFee();
+  console.log(`Flip Fee: ${flipFee} wei`);
+
+  console.log("\n3. Sending request to flip coin...");
+
+  let sequenceNumber: bigint;
+  coinFlipContract.watchEvent.FlipRequest({
+    onLogs: (logs) =>
+      logs.forEach((log) => {
+        console.log(`Flip Request Number/ Sequence Number: ${log.args.sequenceNumber}`);
+        sequenceNumber = log.args.sequenceNumber as bigint;
```

Comment:
Is there any reason to cast this to `bigint`?  How is it typed by default?

If it's already typed as `bigint`, we should remove the cast.  If it isn't, we should remove the cast and parse / validate instead i.e. something like this:

```ts
if (typeof log.args.sequenceNumber === "bigint") {
    ...
} else {
    throw new InvalidSequenceNumberError(log.args.sequenceNumber);
}
```